components/script/dom/element.rs

```diff
@@ -3773,6 +3773,13 @@ pub fn set_cross_origin_attribute(element: &Element, value: Option<DOMString>) {
     }
 }
 
+pub fn get_referrer_policy(element: &Element) -> DOMString {
+    element
+        .get_attribute_by_name(DOMString::from_string(String::from("referrerpolicy")))
+        .map(|attribute: DomRoot<Attr>| attribute.Value())
+        .unwrap_or_else(DOMString::new)
+}
```

Comment:
In order to have consistent naming as other functions, maybe it's good to name this function as `reflect_referrer_policy_attribute`.

Besides, as indicated by the [`spec` of `referrerpolicy` IDL attribute](https://html.spec.whatwg.org/multipage/#dom-script-referrerpolicy), it's `limited to only known values` so we might want to do something like the `reflect_cross_origin_attribute` to filter out those **invalid** values and return a proper value.

https://github.com/servo/servo/blob/9a052a7fab2f169e86a7ea45be49fe405ce177e9/components/script/dom/element.rs#L3754-L3765

In the case of `referrerpolicy`, all valid values are defined in [`#referrer-policy` section in `referrer policy` spec](https://w3c.github.io/webappsec-referrer-policy/#referrer-policy) so we should only reflect those valid values.

Luckily, we've had `determine_policy_for_token` in `components/script/dom/document.rs` to parse the string token to see if it's a valid referrer policy.

Thus, I think we can use the `determine_policy_for_token` function to parse the attribute; if it's a `Some`, then we can return the attribute in `DOMString`; if it's a `None`, then we'll need to return the default value which is an `empty string` (this is defined in [`#referrer-policy-attribute` in html spec](https://html.spec.whatwg.org/multipage/#referrer-policy-attribute).

Reply:
Hmm... I just noticed that `determine_policy_for_token` will also parse some legacy values for the https://html.spec.whatwg.org/multipage/#meta-referrer . @jdm would you think it's still good to use the function if it includes parsing for legacy values 🤔?

Reply:
added `determine_policy_for_token` call for validation.